### PR TITLE
Update docs for Elastic CI Stack v6

### PR DESCRIPTION
**Pull request by triarius**
The main changes are Amazon Linux 2023 and several parameter renames. We are also renaming the `master` branch to `main`. This affects several URLs to github, but also URLs to the stack templates. The github URLs have been changed to be agnostic to what the default branch is.

See https://github.com/buildkite/elastic-ci-stack-for-aws/releases/tag/v6.0.0 for more details.

That link won't be up while this is a draft. In the meantime, use https://github.com/buildkite/elastic-ci-stack-for-aws/releases/tag/v6.0.0-beta1 and https://github.com/buildkite/elastic-ci-stack-for-aws/releases/tag/v6.0.0-beta2 instead.

### CODE REVIEW

**Comment by mbelton-buildkite** [comment]
(I see this is in draft so will ignore for now. Please mention me when you're ready for review)

**Comment by triarius** [comment]
@mbelton-buildkite v6 is out now: https://github.com/buildkite/elastic-ci-stack-for-aws/releases/tag/v6.0.0

File: pages/agent/v3/aws.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ the [GitHub repository](https://github.com/buildkite/elastic-ci-stack-for-aws).
 To run the agent on your own AWS instances, use the installer that matches your
 instance operating system:
 
-* For Amazon Linux 2 use the [Red Hat/CentOS installer](/docs/agent/v3/redhat)
+* For Amazon Linux 2 or later use the [Red Hat/CentOS installer](/docs/agent/v3/redhat)
```
Comment (mbelton-buildkite) [inline pages/agent/v3/aws.md:31]: Missing comma:

```suggestion
* For Amazon Linux 2 or later, use the [Red Hat/CentOS installer](/docs/agent/v3/redhat)
```

File: pages/agent/v3/elastic_ci_aws/managing_elastic_ci_stack.md [hunk 2]
```diff
@@ -87,14 +87,15 @@ or a specific release available from the [releases page](https://github.com/buil
 The latest stable release can be deployed to any of our supported AWS Regions.
 
 The most recent build of the CloudFormation stack is published to
-`https://s3.amazonaws.com/buildkite-aws-stack/master/aws-stack.yml`, along with
+`https://s3.amazonaws.com/buildkite-aws-stack/main/aws-stack.yml`, along with
 a version for each commit at
+`https://s3.amazonaws.com/buildkite-aws-stack/main/${COMMIT}.aws-stack.yml`.
+For commits that are ancestors of , the is build was published to
```
Comment (mbelton-buildkite) [inline pages/agent/v3/elastic_ci_aws/managing_elastic_ci_stack.md:93]: There's something missing here. Ancestors of what?

Comment (triarius) [reply pages/agent/v3/elastic_ci_aws/managing_elastic_ci_stack.md]: Fixed @mbelton-buildkite, though I wonder if this should remain in plain text or some special info box is needed. TBF, I don't think we need it to emphasise it that much, and as we release more versions of the stack, this becomes less important.

File: pages/agent/v3/elastic_ci_aws/managing_elastic_ci_stack.md [hunk 3]
```diff
@@ -87,14 +87,14 @@ or a specific release available from the [releases page](https://github.com/buil
 The latest stable release can be deployed to any of our supported AWS Regions.
 
 The most recent build of the CloudFormation stack is published to
-`https://s3.amazonaws.com/buildkite-aws-stack/master/aws-stack.yml`, along with
+`https://s3.amazonaws.com/buildkite-aws-stack/main/aws-stack.yml`, along with
 a version for each commit at
-`https://s3.amazonaws.com/buildkite-aws-stack/master/${COMMIT}.aws-stack.yml`.
+`https://s3.amazonaws.com/buildkite-aws-stack/main/${COMMIT}.aws-stack.yml`.
+Note: for commits that are ancestors of [419f271b54802c4c8301730bc35b34ed379074c4](https://github.com/buildkite/elastic-ci-stack-for-aws/commit/419f271b54802c4c8301730bc35b34ed379074c4), the build was published to `https://s3.amazonaws.com/buildkite-aws-stack/master/${COMMIT}.aws-stack.yml`.
```
Comment (mbelton-buildkite) [inline pages/agent/v3/elastic_ci_aws/managing_elastic_ci_stack.md:93]: Like you mentioned, this formatting feels a bit awkward. I think it's okay to leave as-is, but I'd probably make it a proper note:

```suggestion

>📘 Ancestors of commit 419f271
> For commits that are ancestors of [419f271](https://github.com/buildkite/elastic-ci-stack-for-aws/commit/419f271b54802c4c8301730bc35b34ed379074c4), the build was published to `https://s3.amazonaws.com/buildkite-aws-stack/master/${COMMIT}.aws-stack.yml`.
```

I also used the shortened commit hash to make it easier to read.

An alternative you could consider that might not draw quite as much attention would be an expand:

```suggestion

<details>
  <summary> 📝 Ancestors of commit 419f271</summary>
  <p>For commits that are ancestors of <a href="https://github.com/buildkite/elastic-ci-stack-for-aws/commit/419f271b54802c4c8301730bc35b34ed379074c4">419f271</a>, the build was published to:
  <p><pre>https://s3.amazonaws.com/buildkite-aws-stack/master/${COMMIT}.aws-stack.yml</pre></p>
</details>
```

I think I'd go with the standard note format.

File: pages/agent/v3/elastic_ci_aws/managing_elastic_ci_stack.md [hunk 4]
```diff
@@ -87,14 +87,14 @@ or a specific release available from the [releases page](https://github.com/buil
 The latest stable release can be deployed to any of our supported AWS Regions.
 
 The most recent build of the CloudFormation stack is published to
-`https://s3.amazonaws.com/buildkite-aws-stack/master/aws-stack.yml`, along with
+`https://s3.amazonaws.com/buildkite-aws-stack/main/aws-stack.yml`, along with
 a version for each commit at
-`https://s3.amazonaws.com/buildkite-aws-stack/master/${COMMIT}.aws-stack.yml`.
+`https://s3.amazonaws.com/buildkite-aws-stack/main/${COMMIT}.aws-stack.yml`.
```
Comment (mbelton-buildkite) [inline pages/agent/v3/elastic_ci_aws/managing_elastic_ci_stack.md:92]: Not blocking: The formatting here also seems a bit awkward with these long URLs as part of the paragraph. I wonder if it would be a bit nicer to split them up, like:

````md
The most recent build of the CloudFormation stack is published to:

```text
https://s3.amazonaws.com/buildkite-aws-stack/main/aws-stack.yml
```

With a version for each commit also published at:

```text
https://s3.amazonaws.com/buildkite-aws-stack/main/${COMMIT}.aws-stack.yml
```
````

That makes them simpler to copy, too.

**Review by mbelton-buildkite — APPROVED** [review]
A couple of suggestions to consider, but happy for you to decide what's best.

File: pages/agent/v3/elastic_ci_aws/managing_elastic_ci_stack.md [hunk 5]
```diff
@@ -86,15 +86,27 @@ or a specific release available from the [releases page](https://github.com/buil
...
+```
+
+With a version for each commit also published at:
+
+```text
+https://s3.amazonaws.com/buildkite-aws-stack/main/${COMMIT}.aws-stack.yml
+```
+
+>📘 Versions prior to v6.0.0
+> Per commit builds for versions prior to v6.0.0, in particular for commits that are ancestors of [419f271](https://github.com/buildkite/elastic-ci-stack-for-aws/commit/419f271b54802c4c8301730bc35b34ed379074c4), were published to
```
Comment (mbelton-buildkite) [inline pages/agent/v3/elastic_ci_aws/managing_elastic_ci_stack.md:102]: Missing hyphen and colon:

```suggestion
> Per-commit builds for versions prior to v6.0.0, in particular for commits that are ancestors of [419f271](https://github.com/buildkite/elastic-ci-stack-for-aws/commit/419f271b54802c4c8301730bc35b34ed379074c4), were published to:
```

File: pages/agent/v3/elastic_ci_aws/managing_elastic_ci_stack.md [hunk 6]
```diff
@@ -86,15 +86,27 @@ or a specific release available from the [releases page](https://github.com/buil
...
+> ```text
```
Comment (mbelton-buildkite) [inline pages/agent/v3/elastic_ci_aws/managing_elastic_ci_stack.md:103]: It looks like this formatting doesn't render correctly:
Screenshot 2023-07-31 at 4 06 18 pm

Adding a blank line seems to solve it:
Screenshot 2023-07-31 at 4 05 53 pm

```suggestion
>
> ```text
```

File: pages/agent/v3/elastic_ci_aws/managing_elastic_ci_stack.md [hunk 7]
```diff
@@ -86,15 +86,27 @@ or a specific release available from the [releases page](https://github.com/buil
...
+> https://s3.amazonaws.com/buildkite-aws-stack/master/${COMMIT}.aws-stack.yml`.
```
Comment (mbelton-buildkite) [inline pages/agent/v3/elastic_ci_aws/managing_elastic_ci_stack.md:104]: Unnecessary formatting:

```suggestion
> https://s3.amazonaws.com/buildkite-aws-stack/master/${COMMIT}.aws-stack.yml
```

**Review by mbelton-buildkite — APPROVED** [review]
